src/runner/PlutoRunner/src/evaluation/deleting globals.jl

```diff
@@ -200,7 +200,16 @@ function try_delete_toplevel_methods(workspace::Module, (cell_id, name_parts)::T
     end
 end
 
-const alive_world_val = methods(Base.sqrt).ms[1].deleted_world # typemax(UInt) in Julia v1.3, Int(-1) in Julia 1.0
+const alive_world_val = typemax(UInt) # This is true at least for julia 1.10 and 1.11, and it's not applicable for julia 1.12. See issue https://github.com/fonsp/Pluto.jl/issues/3259 for more details.
+
+
+# Check if a method has already been deleted/disabled
+is_method_deleted(method::Method) = @static if VERSION < v"1.12.0-beta4"
+    method.deleted_world !== alive_world_val
+else
+    # The dispatch status is set to 0 when a method is deleted. See the file `src/gc.f` changes in PR https://github.com/JuliaLang/julia/pull/58291 for more details.
+    method.dispatch_status === 0x0
```

Comment:
This should be `== 0` right?

```
➜  ~ julia +1.12
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.12.0-beta4 (2025-06-05)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org release
|__/                   |

julia> methods(sqrt)[1].dispatch_status
3

julia> Base.sqrt(::Missing) = "asdffdsf"

julia> Base.delete_method(methods(sqrt)[1])

julia> methods(sqrt)[1].dispatch_status
0

julia> methods(sqrt)[1].dispatch_status === 0x0
false

julia> methods(sqrt)[1].dispatch_status == 0
true
```

I wonder why the tests still passed?

Reply:
The doc removal test from https://github.com/fonsp/Pluto.jl/pull/2096/files#diff-201ebb8d66a94cdf8763d8e046a0fc9c1cc5c76a6fd1725d331a914dbbc1d232R811 should have failed... weird! Indeed the original issue only caused a test failure because precompilation failed with the `const alive_world_val = ...`

Reply:
There was indeed a broken (but not failed) test due to that!
https://github.com/fonsp/Pluto.jl/actions/runs/15752484620/job/44400344732?pr=3270#step:5:1503

Reply:
Sooo what's the conclusion? All good?